src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1670 @@
+            visual_outputs = self.ln_post(visual_outputs)
+        else:
+            visual_outputs_stack = []
+            for visual_output, ln in zip(visual_outputs, self.ln_separate):
+                visual_output = ln(visual_output)
+                visual_outputs_stack.append(visual_output)
+            # shape = [layers, *, width, grid ** 2]
+            visual_outputs = torch.stack(visual_outputs_stack, dim=0)
+        return visual_outputs
+
+    def forward_pre(self, x: torch.Tensor):
+        # shape = [*, width, grid, grid]
+        visual_outputs_pre = self.conv1(x)
+        # shape = [*, width, grid ** 2]
+        visual_outputs_pre = visual_outputs_pre.reshape(visual_outputs_pre.shape[0], visual_outputs_pre.shape[1], -1)
+        # shape = [*, grid ** 2, width]
+        visual_outputs_pre = visual_outputs_pre.permute(0, 2, 1)
+        embeddings_to = self.class_embedding.to(visual_outputs_pre.dtype) + torch.zeros(
+            visual_outputs_pre.shape[0],
+            1,
+            visual_outputs_pre.shape[-1],
+            dtype=visual_outputs_pre.dtype,
+            device=visual_outputs_pre.device,
+        )
+        # shape = [*, grid ** 2 + 1, width]
+        visual_outputs_pre = torch.cat([embeddings_to, visual_outputs_pre], dim=1)
+        visual_outputs_pre = visual_outputs_pre + self.positional_embedding.to(visual_outputs_pre.dtype)
+        visual_outputs_pre = self.ln_pre(visual_outputs_pre)
+        # NLD -> LND
+        visual_outputs_pre = visual_outputs_pre.permute(1, 0, 2)
+        return visual_outputs_pre
+
+    def forward_post(self, x: torch.Tensor):
+        visual_output_post = x.permute(1, 0, 2)
+        visual_output_post = self.ln_post(visual_output_post)
+        return visual_output_post
+
+
+class BridgeTowerCLIP(nn.Module):
+    config_class = BridgeTowerVisionConfig
+
+    def __init__(
+        self,
+        embed_dim: int,
+        # vision
+        input_resolution: int,
+        layers: Union[Tuple[int, int, int, int], int],
+        width: int,
+        patch_size: int,
+        transformer_width: int,
+        output_resolution=224,
+        model_type="bridgetower",
+        stop_gradient=False,
+        vit_layernorm_shared=True,
+        vit_remove_last=False,
+    ):
+        super().__init__()
+
+        vision_heads = width // 64
+        self.visual = BridgeTowerVisualTransformer(
+            input_resolution=input_resolution,
+            patch_size=patch_size,
+            width=width,
+            layers=layers,
+            heads=vision_heads,
+            embed_dim=embed_dim,
+            output_resolution=output_resolution,
+            model_type=model_type,
+            stop_gradient=stop_gradient,
+            vit_layernorm_shared=vit_layernorm_shared,
+            vit_remove_last=vit_remove_last,
+        )
+
+        self.ln_final = BridgeTowerLayerNorm(transformer_width)
+
+        self.initialize_parameters()
+
+    def initialize_parameters(self):
```

Comment:
This should be defined in the `_init_weights` method of `BridgeTowerPreTrainedModel`

Reply:
We have addressed this in our latest commits. Thank you!